src/main/java/org/broadinstitute/barclay/argparser/LegacyCommandLineArgumentParser.java

```diff
@@ -423,13 +423,16 @@ private boolean checkNumArguments() {
                 }
                 if (optionDefinition.isCollection) {
                     final Collection<?> c = (Collection<?>) optionDefinition.field.get(optionDefinition.parent);
+                    if (c.size() == 0 && !optionDefinition.optional) {
+                        messageStream.print("ERROR: Option '" + fullName + "' is required");
+                        return false;
+                    }
                     if (c.size() < optionDefinition.minElements) {
                         messageStream.println("ERROR: Option '" + fullName + "' must be specified at least " +
                                 optionDefinition.minElements + " times.");
                         return false;
                     }
-                } else if (!optionDefinition.optional && !optionDefinition.hasBeenSet &&
-                        !optionDefinition.hasBeenSetFromParent && mutextOptionNames.length() == 0) {
+                } else if (!optionDefinition.optional && !optionDefinition.hasBeenSet && mutextOptionNames.length() == 0) {
```

Comment:
`length()` -> `isEmpty()`

Reply:
mutextOptionNames isn't a collection, its a StringBuilder. No isEmpty on that one.